app/src/main/java/org/oppia/app/utility/FontScaleConfigurationUtil.kt

```diff
@@ -7,29 +7,29 @@ import android.view.WindowManager
 import org.oppia.app.model.StoryTextSize
 import javax.inject.Inject
 
-/** FontScaleConfigurationUtil helps to update font scale. */
-class FontScaleConfigurationUtil @Inject constructor(
-  private val context: Context, private val storyTextSize: String
-) {
+/** Utility to change the scale of font for the entire app. */
+class FontScaleConfigurationUtil @Inject constructor() {
   /** This method updates font scale. */
-  fun adjustFontScale() {
+  fun adjustFontScale(
+    context: Context, storyTextSize: String
+  ) {
     val configuration = context.resources.configuration
-    configuration.fontScale = getFontScaleByStoryTextSize()
+    configuration.fontScale = getStoryTextSizeConfigurationUtil(storyTextSize)
     val metrics: DisplayMetrics = context.resources.displayMetrics
-    val windowManager = context.getSystemService(WINDOW_SERVICE) as WindowManager?
+    val windowManager = context.getSystemService(WINDOW_SERVICE) as? WindowManager
     windowManager!!.defaultDisplay.getMetrics(metrics)
     metrics.scaledDensity = configuration.fontScale * metrics.density
     context.createConfigurationContext(configuration)
     context.resources.displayMetrics.setTo(metrics)
   }
 
   /** This method returns font scale by story text size. */
-  private fun getFontScaleByStoryTextSize(): Float {
-    return when (this.storyTextSize) {
+  private fun getStoryTextSizeConfigurationUtil(storyTextSize: String): Float {
+    return when (storyTextSize) {
       StoryTextSize.SMALL_TEXT_SIZE.name -> .8f
       StoryTextSize.MEDIUM_TEXT_SIZE.name -> 1.0f
-      StoryTextSize.LARGE_TEXT_SIZE.name -> 1.4f
-      else -> 1.6f
+      StoryTextSize.LARGE_TEXT_SIZE.name -> 1.2f
+      else -> 1.4f
```

Comment:
Is this supposed to be EXTRA_LARGE? Why default to the largest value? If we're missing any constants, I suggest adding all constants here and keeping the 'else' case as a reasonable default (like 1.0f).

Reply:
Updated.